src/global/right-click/heroes/dimensional-anchor.js

```diff
@@ -0,0 +1,261 @@
+        allHexes.add(hex); // dimensional anchor applies to ships *in* or adjacent to
+
+        const adjNeighbor = new AdjacencyNeighbor(hex);
+        for (const adjHex of adjNeighbor.getAdjacent()) {
+            allHexes.add(adjHex);
+        }
+
+        const adjWormhole = new AdjacencyWormhole(hex, player.getSlot());
+        for (const adjHex of adjWormhole.getAdjacent()) {
+            allHexes.add(adjHex);
+        }
+
+        const adjHyperlane = new AdjacencyHyperlane(hex);
+        for (const adjHex of adjHyperlane.getAdjacent()) {
+            allHexes.add(adjHex);
+        }
+    }
+
+    // Remove any hexes that do not contain a system tile.
+    const trueHexes = new Set();
+    allHexes.forEach((hex) => {
+        if (hexSet.has(hex)) {
+            trueHexes.add(hex);
+        }
+    });
+
+    // get all non-fighter ships in those systems not owned by the player
+    // whose hero this is
+    const plastic = UnitPlastic.getAll().filter((plastic) =>
+        trueHexes.has(plastic.hex)
+    );
+
+    const ships = plastic.filter((plastic) => {
+        return (
+            SHIPS.includes(plastic.unit) &&
+            plastic.owningPlayerSlot !== cardOwnerSlot
+        );
+    });
+
+    // roll grav rift for each ship
+    const diceObjects = [];
+    ships.forEach((unit) => {
+        const yaw = Math.random() * 360;
+        const d = new Vector(5, 0, 10).rotateAngleAxis(yaw, [0, 0, 1]);
+        const pos = player.getCursorPosition().add(d);
+        const die = new SimpleDieBuilder()
+            .setDeleteAfterSeconds(DELETE_DIE_AFTER_N_SECONDS)
+            .setHitValue(DIMENSIONAL_TEAR_MISS)
+            .setSpawnPosition(pos)
+            .build(player);
+        diceObjects.push([die, unit, 0]);
+    });
+
+    FancyRollGroup.roll(diceObjects, (diceObjects) => {
+        dimensionalAnchorRollFinished(diceObjects, player);
+    });
+
+    // purge the card
+    let purgeContainer = null;
+    for (const obj of world.getAllObjects()) {
+        if (obj.getName() === locale(PURGE_CONTAINER_NAME)) {
+            purgeContainer = obj;
+            break;
+        }
+    }
+
+    if (purgeContainer) {
+        Broadcast.broadcastAll(
+            locale("ui.message.purge", { objectName: locale(cardName) })
+        );
+        purgeContainer.addObjects([card], true);
+    }
+}
+
+function dimensionalAnchorRollFinished(diceObjects, player) {
+    const playerSlot = player.getSlot();
+    const playerDesk = world.TI4.getPlayerDeskByPlayerSlot(playerSlot);
+    const deskName = playerDesk ? playerDesk.colorName : player.getName();
+    const faction = world.TI4.getFactionByPlayerSlot(playerSlot);
+    const playerName = faction ? faction.nameFull : deskName;
+    const color = playerDesk ? playerDesk.color : player.getPlayerColor();
+
+    const parts = [];
+    for (const [die, unit, index] of diceObjects) {
+        assert(typeof index === "number");
+        AutoGravRiftRoller.addLines(die, unit);
```

Comment:
Maybe we should update the `addLines` method to keep them forever until the object get an onGrab event?  The red lines linger a good while so this is probably fine, but it would be a shame if they dropped before the player was finished dealing with units.